staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go

```diff
@@ -65,16 +65,42 @@ func (s *SelectionPredicate) MatchesLabelsAndFields(l labels.Set, f fields.Set)
 	return matched
 }
 
+// TODO: should be namespace.name
+const matchesSingleField = "metadata.name"
```

Comment:
It's wrong - "metadata.name" *is* the correct field (`{"metadata":{"name": ...}}`).  Strip it and this LGTM